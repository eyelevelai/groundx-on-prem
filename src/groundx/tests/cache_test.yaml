suite: "caches render as expected"
chart:
  appVersion: 0.1.0
  version: 0.1.0

tests:
  - it: "default: cache"
    templates:
      - ../templates/services/cache.yaml
      - ../templates/services/cache-metrics.yaml
    values:
      - ../values.yaml
    asserts:
      - matchSnapshot: {}
  - it: "empty: cache"
    templates:
      - ../templates/services/cache.yaml
      - ../templates/services/cache-metrics.yaml
    values:
      - ./files/values.empty.yaml
    asserts:
      - matchSnapshot: {}
  - it: "aws: cache"
    templates:
      - ../templates/services/cache.yaml
      - ../templates/services/cache-metrics.yaml
    values:
      - ../values/values.aws.services.yaml
    asserts:
      - matchSnapshot: {}
  - it: "openshift: cache"
    templates:
      - ../templates/services/cache.yaml
      - ../templates/services/cache-metrics.yaml
    values:
      - ../values/values.openshift.yaml
    asserts:
      - matchSnapshot: {}
