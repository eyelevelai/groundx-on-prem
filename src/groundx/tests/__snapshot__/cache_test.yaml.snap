cache-metrics.yaml:
  1: |
    apiVersion: apps/v1
    kind: StatefulSet
    metadata:
      labels:
        app: cache-metrics
        appVersion: 0.1.0
        chart: groundx-0.1.0
        heritage: Helm
        version: 0.1.0
      name: cache-metrics
      namespace: eyelevel
    spec:
      replicas: 1
      selector:
        matchLabels:
          app: cache-metrics
      serviceName: cache-metrics
      template:
        metadata:
          labels:
            app: cache-metrics
        spec:
          containers:
            - image: public.ecr.aws/c9r4x6y5/eyelevel/redis:latest
              imagePullPolicy: Always
              name: cache-metrics
              ports:
                - containerPort: 6379
                  protocol: TCP
              resources:
                requests:
                  cpu: 0.2
                  memory: 512Mi
              securityContext:
                runAsGroup: 1001
                runAsNonRoot: true
                runAsUser: 1001
                seccompProfile:
                  type: RuntimeDefault
              volumeMounts:
                - mountPath: /mnt/redis
                  name: cache-metrics-persistent-storage
          nodeSelector:
            node: eyelevel-cpu-only
          securityContext:
            runAsGroup: 1001
            runAsNonRoot: true
            runAsUser: 1001
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: cache-metrics-sa
          tolerations:
            - effect: NoSchedule
              key: node
              value: eyelevel-cpu-only
          volumes:
            - emptyDir: {}
              name: cache-metrics-persistent-storage
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: RoleBinding
    metadata:
      name: cache-metrics-sa-rolebinding
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: edit
    subjects:
      - kind: ServiceAccount
        name: cache-metrics-sa
        namespace: eyelevel
  3: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app: cache-metrics
      name: cache-metrics
      namespace: eyelevel
    spec:
      ports:
        - port: 6379
          protocol: TCP
          targetPort: 6379
      selector:
        app: cache-metrics
      type: ClusterIP
  4: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      name: cache-metrics-sa
cache.yaml:
  1: |
    apiVersion: apps/v1
    kind: StatefulSet
    metadata:
      labels:
        app: cache
        appVersion: 0.1.0
        chart: groundx-0.1.0
        heritage: Helm
        version: 0.1.0
      name: cache
      namespace: eyelevel
    spec:
      replicas: 1
      selector:
        matchLabels:
          app: cache
      serviceName: cache
      template:
        metadata:
          labels:
            app: cache
        spec:
          containers:
            - image: public.ecr.aws/c9r4x6y5/eyelevel/redis:latest
              imagePullPolicy: Always
              name: cache
              ports:
                - containerPort: 6379
                  protocol: TCP
              resources:
                requests:
                  cpu: 0.2
                  memory: 512Mi
              securityContext:
                runAsGroup: 1001
                runAsNonRoot: true
                runAsUser: 1001
                seccompProfile:
                  type: RuntimeDefault
              volumeMounts:
                - mountPath: /mnt/redis
                  name: cache-persistent-storage
          nodeSelector:
            node: eyelevel-cpu-only
          securityContext:
            runAsGroup: 1001
            runAsNonRoot: true
            runAsUser: 1001
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: cache-sa
          tolerations:
            - effect: NoSchedule
              key: node
              value: eyelevel-cpu-only
          volumes:
            - emptyDir: {}
              name: cache-persistent-storage
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: RoleBinding
    metadata:
      name: cache-sa-rolebinding
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: edit
    subjects:
      - kind: ServiceAccount
        name: cache-sa
        namespace: eyelevel
  3: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app: cache
      name: cache
      namespace: eyelevel
    spec:
      ports:
        - port: 6379
          protocol: TCP
          targetPort: 6379
      selector:
        app: cache
      type: ClusterIP
  4: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      name: cache-sa
